x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/DocumentPermissions.java

```diff
@@ -36,7 +39,7 @@
  * The document level permissions may be limited by another set of queries in that case the limited
  * queries are used as an additional filter.
  */
-public final class DocumentPermissions {
+public final class DocumentPermissions implements CacheKey {
     private final Set<BytesReference> queries;
     private final Set<BytesReference> limitedByQueries;
```

Comment:
```suggestion
    private final SortedSet<BytesReference> queries;
    private final SortedSet<BytesReference> limitedByQueries;
```

Since we care about the order (for `CacheKey`), I think we should change the type here. A comment might come in handy too.

Reply:
Yes we should. I changed the type to `TreeSet` at instantiation time. But should have also changed the type declaration.